plugin/src/main/kotlin/com/ritense/plugin/PluginDeploymentListener.kt

```diff
@@ -41,40 +45,87 @@ class PluginDeploymentListener(
-                            PluginActionDefinitionId(
-                                actionAnnotation.key,
-                                pluginDefinition
-                            ),
-                            actionAnnotation.title,
-                            actionAnnotation.description,
-                            method.name,
-                            actionAnnotation.activityTypes.toList()
-                        )
-                    )
-                }
             } catch (e: Exception) {
                 throw PluginDefinitionNotDeployedException(pluginAnnotation.key, clazz.name, e)
             }
         }
     }
 
+    private fun createPluginDefinition(clazz: Class<*>, pluginAnnotation: Plugin): PluginDefinition {
+        val pluginProperties: MutableSet<PluginProperty> = mutableSetOf()
+        val pluginDefinition = PluginDefinition(
+            pluginAnnotation.key,
+            pluginAnnotation.title,
+            pluginAnnotation.description,
+            clazz.name,
+            pluginProperties
+        )
+
+        createProperties(pluginDefinition, pluginProperties, clazz)
+
+        return deployPluginDefinition(pluginDefinition)
+    }
+
+    private fun createProperties(
+        pluginDefinition: PluginDefinition,
+        pluginProperties: MutableSet<PluginProperty>,
+        clazz: Class<*>
+    ) {
+        val properties = findPluginProperties(clazz)
+        properties.forEach { (field, propertyAnnotation) ->
+            pluginProperties.add(
+                PluginProperty(
+                    PluginPropertyId(
+                        propertyAnnotation.key,
+                        pluginDefinition
+                    ),
+                    propertyAnnotation.title,
+                    propertyAnnotation.required,
+                    field.name,
+                    field.type.typeName
+                )
+            )
+        }
+    }
+
+    private fun createActionDefinition(deployedPluginDefinition: PluginDefinition, clazz: Class<*>) {
+        val actions = findPluginActions(clazz)
+        actions.forEach { (method, actionAnnotation) ->
+            deployActionDefinition(
+                PluginActionDefinition(
+                    PluginActionDefinitionId(
+                        actionAnnotation.key,
+                        deployedPluginDefinition
+                    ),
+                    actionAnnotation.title,
+                    actionAnnotation.description,
+                    method.name,
+                    actionAnnotation.activityTypes.toList()
+                )
+            )
+        }
+    }
+
     private fun findPluginClasses() : Map<Class<*>, Plugin> {
         return pluginDefinitionResolver.findPluginClasses()
     }
 
     private fun deployPluginDefinition(pluginDefinition: PluginDefinition): PluginDefinition {
         logger.info { "Deploying plugin ${pluginDefinition.key}" }
+        logger.debug { "$pluginDefinition" }
```

Comment:
What does this output look like? is it a proper toString, or does it only output the type+hash like this?